src/nvim/shada.c

```diff
@@ -21,6 +13,11 @@
 #include "nvim/os/os.h"
 #include "nvim/os/time.h"
 #include "nvim/vim.h"
+#ifdef HAVE_BE64TOH
+# define _BSD_SOURCE 1
+# define _DEFAULT_SOURCE 1
+# include <endian.h>
+#endif
```

Comment:
And I would say it is better placed after `nvim` defines list, not break this large group into pieces. Or be in the same group as `vim.h` above: `vim.h+endian.h`, blank, `os/os.h+…+lib/kvec.h`.